src/main/asciidoc/sessions-and-transactions.adoc

```diff
@@ -320,11 +320,45 @@ include::{dotnet-examples}/ExamplesAsync.cs[tags=async-transaction-function]
 include::{dotnet-examples}/ExamplesAsync.cs[tags=async-explicit-transaction]
 ----
 
+[IMPORTANT]
+--
+It is always important to close the session object and it is suggested to keep *session.CloseAsync* in a _finally_ block to make sure all resources (such as network connection) obtained by the session will always be cleaned up properly.
+The session *close* method also enforces rolling back of the last uncommitted or failed transaction in this session.
+Thus it is optional to put *tx.CommitAsync* or *tx.RollbackAsync* in a _finally_ block as long as the *close* of session where the transaction is created will be executed in an outer _finally_ block.
+--
+
 ======
 
 [.include-with-java]
 ======
-This functionality is not available with this driver.
+
+.Auto-commit transactions
+[source, java]
+----
+include::{java-examples}/AsyncAutocommitTransactionExample.java[tags=async-autocommit-transaction]
+----
+
+.Transaction functions
+[source, java]
+----
+include::{java-examples}/AsyncTransactionFunctionExample.java[tags=async-transaction-function]
+----
+
+.Explicit transactions
+[source, java]
+----
+include::{java-examples}/AsyncExplicitTransactionExample.java[tags=async-explicit-transaction]
+----
+
+[IMPORTANT]
+--
+It is important to close the session object to make sure all resources (such as network connections) obtained by the session are cleaned up properly.
```

Comment:
@lutovich Could you please read through the text once more to ensure that I didn't change the meaning of it while editing? Thanks!